Project0/Project0/ViewController.swift

```diff
@@ -11,108 +11,129 @@ import UIKit
 final class ViewController: UIViewController {
     
     enum Operation: Int {
-        case Add = 12
-        case Subtract
-        case Multiple
-        case Divide
-        case Percent
+        case add = 12
+        case subtract
+        case multiple
+        case divide
+        case percent
+        case xy
+        case x2
+        case x3
+        case sin
+        case cos
+        case tan
+        case ctan
+        case ey
+    }
+
+    final let valueE = 2.71828182846
+    
+    var numberOnDisplay: Double {
+        get {
+            if let input = displayLabel.text , let result = Double(input){
+            return result
+            }
+            return 0
+        }
+        set {
+            displayLabel.text = newValue.truncatingRemainder(dividingBy: 1.0) == 0 ? String(format: "%.0f", newValue) : String(newValue)
+        }
     }
     
-    var numberOnDisplay = 0.0
     var previosNumber = 0.0
-    var operationDuring = false
-    var operationDone = false
-    var operation = 0
+    var isOperationProgress = false
+    var isOperationDone = false
+    var operation: Int!
     
     @IBOutlet weak var displayLabel: UILabel!
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-    }
-    
+
     @IBAction func acButton(_ sender: Any) {
-        displayLabel.text = "0"
-        previosNumber = 0
-        operation = 0
         numberOnDisplay = 0
-        operationDuring = false
-        operationDone = false
+        previosNumber = 0
+        operation = nil
+        isOperationProgress = false
```

Comment:
isOperationProgress  -> isOperationInProgress is more readable. But still I'm sure that you should delete it.